### PR TITLE
Clarify parallel functions workshop

### PR DESCRIPTION
**Pull request by edgarfinn**
#337
- Added an introduction to the importance of parallel functions
- Updated week schedule to link only to this workshop

### CODE REVIEW

File: coursebook/week-3/morning-challenge.md [hunk 1]
```diff
@@ -1,17 +1,28 @@
 **Author**: [@besarthoxhaj](https://github.com/besarthoxhaj)  
 
 **Maintainer**: TBC
+## Parallel Functions Workshop
```
Comment (jsms90) [inline coursebook/week-3/morning-challenge.md]: It's a bit confusing to call this a "workshop" here, especially when you have a separate heading below for "morning challenge

I suggest changing this line to "Parallel Functions Morning Challenge" and removing [line 11](https://github.com/foundersandcoders/master-reference/pull/408/files#diff-fad2f5d14a0003214b8ad6dac0344de3L5) altogether?

**Comment by jsms90** [comment]
@edgarfinn Could you reference the related issues in the body of your PR please?

File: coursebook/week-3/morning-challenge.md [hunk 2]
```diff
@@ -1,17 +1,26 @@
 **Author**: [@besarthoxhaj](https://github.com/besarthoxhaj)  
 
 **Maintainer**: TBC
+## Parallel Functions Morning Challenge
 
-## Morning challenge - day 3
+When making API calls, exactly how quickly you receive a response can depend on several factors: the speed of your connection, the performance of your code, and the performance of the API that you're calling.
```
Comment (des-des) [inline coursebook/week-3/morning-challenge.md]: The performance of you code will not effect the response time, once you have made a request, you code will not do anything until you get a response.

Connection speed and latency, and the api response time are the two factors that effect the response time

File: coursebook/week-3/morning-challenge.md [hunk 3]
```diff
@@ -1,17 +1,26 @@
...
+Often, you will want to process the response that you receive for an API (like perhaps updating the DOM, or even making more API calls based on the response of the first), but you will probably only want to run that processes ONLY once you've received a response from the API call. Otherwise you may get an error message like ```cannot read property of null```.
+
+This is where parallel functions come in handy....
```
Comment (des-des) [inline coursebook/week-3/morning-challenge.md]: Parallel -> do many async things at the same time
waterfall -> do async things one after the other

File: coursebook/week-3/morning-challenge.md [hunk 4]
```diff
@@ -1,17 +1,26 @@
 **Author**: [@besarthoxhaj](https://github.com/besarthoxhaj)  
 
 **Maintainer**: TBC
+## Parallel Functions Morning Challenge
 
-## Morning challenge - day 3
+When making API calls, exactly how quickly you receive a response can depend on several factors: the speed of your connection, the performance of your code, and the performance of the API that you're calling.
+Often, you will want to process the response that you receive for an API (like perhaps updating the DOM, or even making more API calls based on the response of the first), but you will probably only want to run that processes ONLY once you've received a response from the API call. Otherwise you may get an error message like ```cannot read property of null```.
```
Comment (des-des) [inline coursebook/week-3/morning-challenge.md]: Yeah you are talking about waterfall here

**Comment by edgarfinn** [comment]
how about this?
https://github.com/foundersandcoders/master-reference/pull/408/commits/1bd7d0d949ab9cb8196ec9ddf48fe624df4029ac

**Comment by edgarfinn** [comment]
https://github.com/foundersandcoders/master-reference/pull/408/commits/1bd7d0d949ab9cb8196ec9ddf48fe624df4029ac

File: coursebook/week-3/morning-challenge.md [hunk 5]
```diff
@@ -1,17 +1,26 @@
 **Author**: [@besarthoxhaj](https://github.com/besarthoxhaj)  
 
 **Maintainer**: TBC
+## Parallel Functions Morning Challenge
 
-## Morning challenge - day 3
+When making API calls, exactly how quickly you receive a response can depend on the speed of your connection, and the performance speed of the API that you're calling.
```
Comment (eliascodes) [inline coursebook/week-3/morning-challenge.md]: Suggest "performance speed" is redundant; simply "performance" is fine.

File: coursebook/week-3/morning-challenge.md [hunk 6]
```diff
@@ -1,17 +1,26 @@
...
+When making several API call, or reading data from several files at the same time, the order in which each task returns will depend on how quickly each task can be completed, which may not always be the same order that the tasks were called.
```
Comment (eliascodes) [inline coursebook/week-3/morning-challenge.md]: "...several API **calls**"

Also suggest the use of "returns" here could be misleading. Might it be better to replace it with "...is completed"

File: coursebook/week-3/morning-challenge.md (hunk 5 above)
Comment (des-des) [inline coursebook/week-3/morning-challenge.md]: There are three things I can think of
  1. Connection Speed
  2. Connection Latency
  3. **API response time**

**Comment by edgarfinn** [comment]
@eliascodes I'm not aware of a solution existing for this challenge. This pull request is about clarifying the instructions on master branch. 

A solution would surely need be merged to a solution branch, no?

**Comment by jsms90** [comment]
@eliascodes this was made by Bes before the solutions branch pattern existed. In order to follow the solutions branch pattern, we'd need to move this to its own repo anyway. Or just have an extra file? But maybe sticking with one convention is better?

As far as I know, Bes is not actively maintaining this morning challenge

**Comment by edgarfinn** [comment]
@jsms90 @eliascodes I'm happy to offer a solution if we decide to create a solutin branch on master ref.

**Comment by jsms90** [comment]
@edgarfinn I think it would be really good to have a solution somewhere! So if you don't mind making one, feel free to open an issue and start working against that.

@eliascodes @des-des Thoughts on where to store it? Should we separate this file into its own repo to follow the pattern of all the rest?

**Comment by jsms90** [comment]
@eliascodes @des-des ?

**Comment by eliascodes** [comment]
Could it go in fragments? Then you can have the README with the problem statement and a solution file

**Comment by jsms90** [comment]
As I understand it, `fragments` is for things that don't actually have a place in the curriculum any more / yet e.g. the testing week stopwatch. @des-des ?

@eliascodes Why not put the solution with the rest of the "part" i.e. challenge?

**Comment by des-des** [comment]
am easy

**Pull request merged**